frontend/src/features/env/EnvService.ts

```diff
@@ -1,7 +1,86 @@
-import { ClientEnvVars } from '~shared/types/core'
+import { PublicFormViewDto, switchEnvFeedbackFormBodyDto } from '~shared/types'
+import { ClientEnvVars, SuccessMessageDto } from '~shared/types/core'
 
+import { transformAllIsoStringsToDate } from '~utils/date'
 import { ApiService } from '~services/ApiService'
 
+import { PUBLIC_FORMS_ENDPOINT } from '~features/public-form/PublicFormService'
+
 export const getClientEnvVars = async (): Promise<ClientEnvVars> => {
   return ApiService.get<ClientEnvVars>('/client/env').then(({ data }) => data)
 }
+
+// TODO #4279: Remove after React rollout is complete
+const createFeedbackResponsesArray = (
+  formInputs: switchEnvFeedbackFormBodyDto,
+  feedbackForm: PublicFormViewDto | undefined,
+) => {
+  const responses = []
+  for (const [key, value] of Object.entries(formInputs)) {
+    const entry = {
+      _id:
+        key === 'url'
+          ? feedbackForm?.form.form_fields[0]._id
+          : key === 'feedback'
+          ? feedbackForm?.form.form_fields[1]._id
+          : feedbackForm?.form.form_fields[2]._id,
+      question: key,
+      answer: value ?? '',
+      fieldType:
+        key === 'url'
+          ? feedbackForm?.form.form_fields[0].fieldType
+          : key === 'feedback'
+          ? feedbackForm?.form.form_fields[1].fieldType
+          : feedbackForm?.form.form_fields[2].fieldType,
+    }
+    responses.push(entry)
```

Comment:
Suggestion to reduce code duplication.

```suggestion
    const fieldIndex = key === 'url' ? 0 : key === 'feedback' ? 1 : 2
    const { _id, fieldType } = feedbackForm?.form.form_fields[fieldIndex] ?? {}
    const entry = {
      _id,
      question: key,
      answer: value ?? '',
      fieldType,
    }
    responses.push(entry)
```

Also question: what does it mean for the entries when `feedbackForm` is `undefined`?